pkg/apis/networking/validation/validation.go

```diff
@@ -59,20 +61,39 @@ func ValidateNetworkPolicyName(name string, prefix bool) []string {
 
 // ValidateNetworkPolicyPort validates a NetworkPolicyPort
 func ValidateNetworkPolicyPort(port *networking.NetworkPolicyPort, portPath *field.Path) field.ErrorList {
+
+	allowEndPort := utilfeature.DefaultFeatureGate.Enabled(features.NetworkPolicyEndPort)
+
 	allErrs := field.ErrorList{}
 	if port.Protocol != nil && *port.Protocol != api.ProtocolTCP && *port.Protocol != api.ProtocolUDP && *port.Protocol != api.ProtocolSCTP {
 		allErrs = append(allErrs, field.NotSupported(portPath.Child("protocol"), *port.Protocol, []string{string(api.ProtocolTCP), string(api.ProtocolUDP), string(api.ProtocolSCTP)}))
 	}
+
+	if !allowEndPort && port.EndPort != nil {
+		allErrs = append(allErrs, field.Forbidden(portPath.Child("endPort"), "This field can be enabled with the NetworkPolicyEndPort feature gate"))
+	}
 	if port.Port != nil {
 		if port.Port.Type == intstr.Int {
 			for _, msg := range validation.IsValidPortNum(int(port.Port.IntVal)) {
 				allErrs = append(allErrs, field.Invalid(portPath.Child("port"), port.Port.IntVal, msg))
 			}
+			if allowEndPort && *port.EndPort < port.Port.IntVal {
+				// TODO: Find the right error field
+				allErrs = append(allErrs, field.Invalid(portPath.Child("endPort"), port.Port.IntVal, "endPort field must be equal or bigger than port field"))
+			}
 		} else {
+			if allowEndPort && port.EndPort != nil {
+				// TODO: Find the right error field
+				allErrs = append(allErrs, field.Invalid(portPath.Child("endPort"), port.Port.IntVal, "endPort cannot be used when port is non-numeric"))
```

Comment:
`port.Port.StrVal`

Reply:
Here a question: should this be port.EndPort or port.Port.StrVal? I'm saying that endPort is invalid but printing the value from Port, so wondering if this should be actually the endPort value (or maybe if I can add two error lines: one for the endPort and other for the Port).

Reply:
I've changed to *port.EndPort right now as we're complaining from this being defined and not the named-port. But this can be revisited again :)